resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsAllocator.scala

```diff
@@ -148,6 +162,10 @@ class ExecutorPodsAllocator(
 
   def isDeleted(executorId: String): Boolean = deletedExecutorIds.contains(executorId.toLong)
 
+  private[k8s] def stopApplication(exitCode: Int): Unit = {
+    sys.exit(exitCode)
```

Comment:
Will Kubernetes re-schedule the driver pod in this case?

Reply:
No, AFAIK there is no application attempt mechanism like yarn does

Reply:
>  manually tested in internal K8s cluster.

Which deploy-mode have you tested?

Reply:
cluster mode